tests/test_model_creation_deletion.py

```diff
@@ -58,17 +64,25 @@ def test_deleted_elements_are_removed(model, deletion_target):
     assert len(comps) != 2, "List length did not change"
 
     with pytest.raises(KeyError):
-        model._loader[olduuid]
+        model._loader[olduuid]  # pylint: disable=pointless-statement
 
     assert not model._loader.xpath(
         XPATH_UUID.format(olduuid)
     ), "Element is still present in tree after deleting"
 
 
-def test_delete_all_deletes_matching_objects(model):
+def test_delete_all_deletes_matching_objects(model: capellambse.MelodyModel):
     comps = model.la.root_component.components
     assert len(comps) == 2
 
     comps.delete_all(name="Delete Me")
     assert len(comps) == 1
     assert comps[0].name == "Keep Me"
+
+
+def test_delete_elements_from_lookups(model: capellambse.MelodyModel):
+    assert model.la.all_functions[0] == model.la.root_function
```

Comment:
We should rather `assert` over the entire list here, just to guard against the test model accidentally being changed in the future.

```suggestion
    assert model.la.all_functions == [model.la.root_function]
```